### PR TITLE
Add option to create launch config in workspace file

### LINKED ISSUE #96567: Off to create launch configuration in workspace file
Verifying https://github.com/microsoft/vscode/issues/95482

- create a new workspace with some new empty folders
- save the workspace file
- open the debug viewlet and click on the `create a launch.json file`
- I can choose in which folder to create
- I actually want to store it in the workspace file. Would be cool if that's an option too.

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #96567

### CODE REVIEW

File: src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts [hunk 1]
```diff
@@ -715,10 +717,29 @@ class WorkspaceLaunch extends AbstractLaunch implements ILaunch {
 		return this.configurationService.inspect<IGlobalConfig>('launch').workspaceValue;
 	}
 
-	async openConfigFile(sideBySide: boolean, preserveFocus: boolean): Promise<{ editor: IEditorPane | null, created: boolean }> {
+	async openConfigFile(sideBySide: boolean, preserveFocus: boolean, type?: string, token?: CancellationToken): Promise<{ editor: IEditorPane | null, created: boolean }> {
+		const ws = this.contextService.getWorkspace();
+
+		if (ws.folders.length === 0) {
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts:723]: Is this check necesery?
If I do not have any folders in the workspace I do not even get offered "workspace" in the quick pick

File: src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts [hunk 2]
```diff
@@ -715,10 +717,29 @@ class WorkspaceLaunch extends AbstractLaunch implements ILaunch {
...
+			return { editor: null, created: false };
+		}
+
+		let launchExistInFile = !!this.getConfig();
+		if (!launchExistInFile) {
+			// Launch property in workspace config not found: create one by collecting launch configs from debugConfigProviders
+			let content = '';
+			const adapter = await this.configurationManager.guessDebugger(type);
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts:731]: I see we have similar code in the `Launch`. Might it make sense to extract this code to the `AbstractLaunch` so both clases use it? Not sure if 100% good idea..

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
Thanks a lot for this PR, great work!
I have left some comments in the PR, can you please address them. Also
1) There seems to be a conflict, can you please resolve it (just removing `sideBySide` options should do it)
2) I am not a fan of the current UI. Can we change "workspace" to be "Workspace". Can we change the placeholder message to also mention this new optoin?

Screenshot 2020-05-11 at 13 00 26

File: src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts (hunk 1 above)
Comment (jeanp413) [reply src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts:723]: That's an edge case I found with the following steps:
1. Open a folder `f1`
2. Add to workspace folder `f2`. An untitled workspace will be created
3. Remove folder `f1` and `f2` from the workspace. At this point we have no folders in the workspace.
4. `this.contextService.getWorkbenchState` returns `WorkbenchState.WORKSPACE` (is this expected?) so the `openConfigFile` function gets called.
Thinking more about this I think I should move that check here https://github.com/microsoft/vscode/blob/7a424f606b2b73547a66e12435a4064dc6b37087/src/vs/workbench/contrib/debug/browser/debugActions.ts#L85

Comment (isidorn) [reply src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts:723]: Yeah that state is the empty workspace state - you are stil in a workspace just do not have any folder in the workspace. So you are correct we should cover that case. Thanks

**Comment by jeanp413** [comment]
Pushed some changes addressing the feedback

**Comment by isidorn** [comment]
Thanks a lot for this PR. Merging in. 
I am doing some changes in that file at the moment so I might polish it a bit on top.
☀️

**Pull request merged**